### PR TITLE
Add `max` and `min` functions for mal down sampling

### PR DESCRIPTION
**Pull request by peachisai**
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

# User Case
```
metricsRules:
  - name: consumer_latency
    exp: rocketmq_group_get_latency_by_storetime.sum(['cluster','topic','group']).downsampling(MAX)
```
The opentelemetry metrics
```
rocketmq_consumer_message_size{cluster="DefaultCluster",broker="broker-a",topic="topic1",group="group2",} 21350.0
rocketmq_consumer_message_size{cluster="DefaultCluster",broker="broker-b",topic="topic1",group="group2",} 7700.0
```
 The ```sum``` aggregate method use 'cluster','topic' and 'group' as aggregate-by params,so I can sum the metrics of the two brokers, and then get the max message size instead of the default ```avg``` function value

The same scenario for ```min``` function

### CODE REVIEW

**Comment by wu-sheng** [comment]
Could you provide the use cases?

**Comment by peachisai** [comment]
> Could you provide the use cases?

I updated the use case

**Comment by peachisai** [comment]
> > > Could you provide the use cases?
> > 
> > 
> > I updated the use case
> 
> Why do you need a max value? I think in most case, percentile should be better to measure latency? Does max bring a special value?

I prefer the maximum value. This looks more intuitive. For example, with size and offset metrics.

**Comment by wu-sheng** [comment]
Size and offset make sense. But about latency, percentile is better.

**Comment by wu-sheng** [comment]
Please update the description for the meaningful use cases.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/function/max/MaxFunction.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    @Override
+    public void accept(MeterEntity entity, Long value) {
+        setEntityId(entity.id());
+        setServiceId(entity.serviceId());
+        if (this.value < value) {
+            setValue(value);
+        }
+    }
+
+    @Entrance
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/function/max/MaxFunction.java:80]: `@Entrance` is misused here, and this error exists in many other MAL functions. This annotation is used in MAL engine use, in MAL, `accept` method is the main entrance by default.

**Comment by peachisai** [comment]
> Please update the description for the meaningful use cases.

Updated

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/function/max/MaxFunction.java (hunk 1 above)
Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/function/max/MaxFunction.java:80]: I am trying to fix existing codes.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/function/max/MaxFunction.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+     * Service ID is required for sort query.
+     */
+    @Setter
+    @Getter
+    @Column(name = InstanceTraffic.SERVICE_ID)
+    private String serviceId;
+
+    @Getter
+    @Setter
+    @Column(name = VALUE, dataType = Column.ValueDataType.COMMON_VALUE, function = Function.Avg)
```
Comment (mrproliu) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/function/max/MaxFunction.java:67]: The function type is confusion. I think the new function types should be define.

Comment (peachisai) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/function/max/MaxFunction.java]: > The function type is confusion. I think the new function types should be define.

I added the new function types. Should I also add new code for this query aggregation?

**Review by mrproliu — APPROVED** [review]

**Pull request merged**